internal/api/v1/gitconfig/create.go

```diff
@@ -0,0 +1,117 @@
+	gitbridge "github.com/epinio/epinio/internal/bridge/git"
+	"github.com/epinio/epinio/internal/cli/server/requestctx"
+	"github.com/epinio/epinio/internal/helmchart"
+	apierror "github.com/epinio/epinio/pkg/api/core/v1/errors"
+	"github.com/epinio/epinio/pkg/api/core/v1/models"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
+
+	"github.com/gin-gonic/gin"
+)
+
+// Create handles the API endpoint /gitconfigs (POST).
+// It creates a gitconfig with the specified name.
+func Create(c *gin.Context) apierror.APIErrors {
+	ctx := c.Request.Context()
+	logger := requestctx.Logger(ctx)
+
+	cluster, err := kubernetes.GetCluster(ctx)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	var request models.GitconfigCreateRequest
+	err = c.BindJSON(&request)
+	if err != nil {
+		return apierror.NewBadRequestError(err.Error())
+	}
+
+	gitconfigName := request.ID
+	if gitconfigName == "" {
+		return apierror.NewBadRequestError("name of gitconfig to create not found")
+	}
+	errorMsgs := validation.IsDNS1123Subdomain(gitconfigName)
+	if len(errorMsgs) > 0 {
+		return apierror.NewBadRequestErrorf("Git configurations' name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name', or '123-abc').")
+	}
+
+	manager, err := gitbridge.NewManager(logger, cluster.Kubectl.CoreV1().Secrets(helmchart.Namespace()))
+	if err != nil {
+		return apierror.InternalError(err, "creating git configuration manager")
+	}
+
+	gitconfigList := manager.Configurations
+
+	// see delete.go
+	gcNames := map[string]struct{}{}
+	for _, gitconfig := range gitconfigList {
+		gcNames[gitconfig.ID] = struct{}{}
+	}
+
+	// already known ?
+	if _, ok := gcNames[gitconfigName]; ok {
+		return apierror.NewConflictError("gitconfig", gitconfigName)
+	}
+
+	secret := gitbridge.NewSecretFromConfiguration(gitbridge.Configuration{
+		ID:          request.ID,
+		URL:         request.URL,
+		Provider:    request.Provider,
+		Username:    request.Username,
+		Password:    request.Password,
+		UserOrg:     request.UserOrg,
+		Repository:  request.Repository,
+		SkipSSL:     request.SkipSSL,
+		Certificate: request.Certificates,
+	})
+
+	err = cluster.CreateSecret(ctx, helmchart.Namespace(), secret)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	err = addGitconfigToUser(ctx, request.ID)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	response.Created(c)
+	return nil
+}
+
+// addGitconfigToUser will add the gitconfig to the User's gitconfigs
+func addGitconfigToUser(ctx context.Context, gitconfig string) error {
+	user := requestctx.User(ctx)
+
+	authService, err := auth.NewAuthServiceFromContext(ctx)
+	if err != nil {
+		return errors.Wrap(err, "error creating auth service")
+	}
+
+	err = authService.AddGitconfigToUser(ctx, user.Username, gitconfig)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error adding gitconfig [%s] to user [%s]",
```

Comment:
NITP: the `Wrapf` can be used to format the error message.

Reply:
I can retry. IIRC when I used that at first the linter claimed that `Wrapf` does not exist.

Reply:
Done with commit f6bc2eca5f02b7f0634afe443b9e2ea734faeb05

Works now. No idea anymore what was going at the time.